### DISCUSSION TITLE
Generating new access_token with help of refresh_token from REACT side.

### DISCUSSION THREAD

**Discussion by bhagya1101** [directus/directus#16798] [Archive]
I have made a login screen in REACT and I want users to Login from there but using Directus.
I used POST method that passes "email", "password" as req.body from REACT SIDE to ' /auth/login ' and I got access_token in response on REACT side.

Now, if the access_token expires after 15 mins, how do I regenerate access_token with help of refresh_token in REACT side ?

Do I have to check everytime for expiry of access_token and generate it with the help of ' /auth/refresh ' ? OR is there any other method for it ?

**Comment by paescuj** [comment]
I recommend using the official [JS SDK](https://docs.directus.io/reference/sdk.html) which will handle the refresh of tokens automatically (see also https://docs.directus.io/reference/sdk.html#constructor).

**Comment by raphourbe** [comment]
Hi,

I have written two posts on the topic. 

- In the first one: [Using Directus js-sdk in a Next.js app with several useSWR calls.](https://rphl.dev/blog/directus-sdk-nextjs-several-useswr), I show how I created an AuthProvider to handle the refresh token using the directus-sdk. I use Next.js but I am sure you can use it elsewhere.
- The second if you use Next.js and next-auth: [Using Directus with Next-Auth to handle authentication in your frontend.](https://rphl.dev/blog/nextauth-directus) In this case, I do not use directus-sdk.

As mentioned in [Directus' documentation](https://docs.directus.io/reference/sdk.html#refresh-auth-token), the directus-sdk handles the refresh part directly, and you can also do it manually with `await directus.auth.refresh();`.

If you do not use the sdk, Directus sends you a `refresh_token` and yes, you will have to manually refresh it, if you do not use tools like next-auth.
